arangod/Aql/AqlValue.h

```diff
@@ -272,6 +275,79 @@ struct AqlValue final {
     } longNumberMeta;
     static_assert(sizeof(longNumberMeta) == 16,
                   "VPACK_INLINE_INT64 layout is not 16 bytes!");
+
+    // VPACK_SUPERVISED_SLICE
+    struct {
+      velocypack::Slice toSlice() const noexcept {
+        return velocypack::Slice(reinterpret_cast<uint8_t const*>(
+            pointer + sizeof(arangodb::ResourceMonitor*)));
+      }
+      uint64_t getLength() const noexcept {
+        return velocypack::Slice(pointer + sizeof(arangodb::ResourceMonitor*))
+            .byteSize();
+      }
```

Comment:
Can we just embed length information in the first 64bit, just like the managed slice does?